server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -2215,9 +2275,41 @@ public String getStorageGroupName() {
     return storageGroupName;
   }
 
+  /**
+   * Check if the data of "tsFileResource" all exist locally by comparing the historical versions
+   * in the partition of "partitionNumber". This is available only when the IoTDB which generated
+   * "tsFileResource" has the same close file policy as the local one.
+   * If one of the version in "tsFileResource" equals to a version of a working file, false is
+   * also returned because "tsFileResource" may have unwritten data of that file.
```

Comment:
Please do not ignore the clause "which generated 'tsFileResource' ".
A TsFile is not always generated by the local IoTDB instance, and it may be generated by another IoTDB and is sent by the sync module or catch-up in the distributed version.